[Bug]: panic in internal/pipe/scoop.dataFor

### What happened?

I did a test release of my project using the GitHub Goreleaser Action with Goreleaser version 0.180.0 and got a panic in Goreleaser. Selected log entries:

```
Downloading https://github.com/goreleaser/goreleaser/releases/download/v0.180.0/goreleaser_Linux_x86_64.tar.gz
Extracting GoReleaser
/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/e7c1e0b9-d798-4a93-85d6-a715d3e2ef75 -f /home/runner/work/_temp/09860fcf-91a4-40bc-a049-7da90b8fa143
GoReleaser latest installed successfully
No tag found for commit 'd72e437c'. Snapshot forced
/opt/hostedtoolcache/goreleaser-action/0.180.0/x64/goreleaser release --skip-publish --snapshot
   • releasing...     
...
   • scoop manifests  
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1321099]

goroutine 8 [running]:
github.com/goreleaser/goreleaser/internal/pipe/scoop.dataFor(0xc000169800, {0x0, 0x0}, {0xc00064ca80, 0x4, 0xc0001899c0})
	github.com/goreleaser/goreleaser@v0.180.0/internal/pipe/scoop/scoop.go:210 +0x1b9
github.com/goreleaser/goreleaser/internal/pipe/scoop.doRun(0xc000169800, {0x0, 0x0})
	github.com/goreleaser/goreleaser@v0.180.0/internal/pipe/scoop/scoop.go:92 +0x28d
github.com/goreleaser/goreleaser/internal/pipe/scoop.Pipe.Run({}, 0x263fe40)
	github.com/goreleaser/goreleaser@v0.180.0/internal/pipe/scoop/scoop.go:43 +0x45
github.com/goreleaser/goreleaser/internal/middleware/errhandler.Handle.func1(0x17ff6d8)
	github.com/goreleaser/goreleaser@v0.180.0/internal/middleware/errhandler/error.go:14 +0x22
github.com/goreleaser/goreleaser/internal/middleware/logging.Log.func1(0x269c318)
	github.com/goreleaser/goreleaser@v0.180.0/internal/middleware/logging/logging.go:34 +0x142
github.com/goreleaser/goreleaser/internal/middleware/skip.Maybe.func1(0x1686600)
	github.com/goreleaser/goreleaser@v0.180.0/internal/middleware/skip/skip.go:29 +0xf0
github.com/goreleaser/goreleaser/cmd.releaseProject.func1()
	github.com/goreleaser/goreleaser@v0.180.0/cmd/release.go:113 +0x11d
github.com/caarlos0/ctrlc.(*Ctrlc).Run.func1()
	github.com/caarlos0/ctrlc@v1.0.0/ctrlc.go:39 +0x29
created by github.com/caarlos0/ctrlc.(*Ctrlc).Run
	github.com/caarlos0/ctrlc@v1.0.0/ctrlc.go:38 +0xaf
Error: The process '/opt/hostedtoolcache/goreleaser-action/0.180.0/x64/goreleaser' failed with exit code 2
```

Links:
* [Full test log](https://github.com/twpayne/chezmoi/runs/3713443818?check_suite_focus=true) (see Build Release step).
* [`.goreleaser.yaml`](https://github.com/twpayne/chezmoi/blob/f42866bc/.goreleaser.yaml)
* [GitHub Action step](https://github.com/twpayne/chezmoi/blob/f42866bcdab0301cb876f1279e9c22400db96307/.github/workflows/main.yml#L133-L137)

### How can we reproduce this?

So far I have only seen this running on GitHub Actions, and have not been able to reproduce this locally.

### goreleaser version

```bash
goreleaser version 0.180.0
commit: 3d62f95fd53bfa037e3e39894eefe637a8758333
built at: 2021-09-24T23:18:52Z
built by: goreleaser
module version: v0.180.0, checksum: h1:FuP0EX2UBAGaXF0xk0Uag2k53qO6cs9IX9NZkGi2Gto=

https://goreleaser.com
```


### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

Awesome project - thank you!

I'll do some more digging to see if I can find out how to reproduce this locally.

--- a/internal/client/client.go
+++ b/internal/client/client.go
@@ -43,20 +43,11 @@ type Client interface {
 
 // New creates a new client depending on the token type.
 func New(ctx *context.Context) (Client, error) {
-	log.WithField("type", ctx.TokenType).Debug("token type")
-	if ctx.TokenType == context.TokenTypeGitHub {
-		return NewGitHub(ctx, ctx.Token)
-	}
-	if ctx.TokenType == context.TokenTypeGitLab {
-		return NewGitLab(ctx, ctx.Token)
-	}
-	if ctx.TokenType == context.TokenTypeGitea {
-		return NewGitea(ctx, ctx.Token)
-	}
-	return nil, nil
+	return newWithToken(ctx, ctx.Token)
 }
 
 func newWithToken(ctx *context.Context, token string) (Client, error) {
+	log.WithField("type", ctx.TokenType).Debug("token type")
 	if ctx.TokenType == context.TokenTypeGitHub {
 		return NewGitHub(ctx, token)
 	}
@@ -66,7 +57,7 @@ func newWithToken(ctx *context.Context, token string) (Client, error) {
 	if ctx.TokenType == context.TokenTypeGitea {
 		return NewGitea(ctx, token)
 	}
-	return nil, nil
+	return nil, fmt.Errorf("invalid client token type: %q", ctx.TokenType)
 }
 
 func NewIfToken(ctx *context.Context, cli Client, token string) (Client, error) {

--- a/internal/client/client_test.go
+++ b/internal/client/client_test.go
@@ -12,7 +12,7 @@ func TestClientEmpty(t *testing.T) {
 	ctx := &context.Context{}
 	client, err := New(ctx)
 	require.Nil(t, client)
-	require.NoError(t, err)
+	require.EqualError(t, err, `invalid client token type: ""`)
 }
 
 func TestClientNewGitea(t *testing.T) {
@@ -153,7 +153,7 @@ func TestNewWithToken(t *testing.T) {
 		}
 
 		cli, err := newWithToken(ctx, "{{ .Env.TK }}")
-		require.NoError(t, err)
+		require.EqualError(t, err, `invalid client token type: "nope"`)
 		require.Nil(t, cli)
 	})
 }

--- a/internal/pipe/brew/brew.go
+++ b/internal/pipe/brew/brew.go
@@ -83,11 +83,6 @@ func (Pipe) Run(ctx *context.Context) error {
 
 // Publish brew formula.
 func (Pipe) Publish(ctx *context.Context) error {
-	// we keep GitHub as default for now, in line with releases
-	if string(ctx.TokenType) == "" {
-		ctx.TokenType = context.TokenTypeGitHub
-	}
-
 	cli, err := client.New(ctx)
 	if err != nil {
 		return err

--- a/internal/pipe/env/env.go
+++ b/internal/pipe/env/env.go
@@ -83,12 +83,6 @@ func (Pipe) Run(ctx *context.Context) error {
 		return err
 	}
 
-	if githubToken != "" {
-		log.Debug("token type: github")
-		ctx.TokenType = context.TokenTypeGitHub
-		ctx.Token = githubToken
-	}
-
 	if gitlabToken != "" {
 		log.Debug("token type: gitlab")
 		ctx.TokenType = context.TokenTypeGitLab
@@ -99,6 +93,15 @@ func (Pipe) Run(ctx *context.Context) error {
 		log.Debug("token type: gitea")
 		ctx.TokenType = context.TokenTypeGitea
 		ctx.Token = giteaToken
+	}
+
+	if githubToken != "" {
+		log.Debug("token type: github")
+		ctx.Token = githubToken
+	}
+
+	if ctx.TokenType == "" {
+		ctx.TokenType = context.TokenTypeGitHub
 	}
 
 	return nil

--- a/internal/pipe/env/env_test.go
+++ b/internal/pipe/env/env_test.go
@@ -59,6 +59,12 @@ func TestSetDefaultTokenFiles(t *testing.T) {
 		})
 		require.EqualError(t, Pipe{}.Run(ctx), `template: tmpl:1: unexpected "}" in operand`)
 	})
+
+	t.Run("no token", func(t *testing.T) {
+		ctx := context.New(config.Project{})
+		require.NoError(t, Pipe{}.Run(ctx))
+		require.Equal(t, ctx.TokenType, context.TokenTypeGitHub)
+	})
 }
 
 func TestValidGithubEnv(t *testing.T) {

